src/shared/components/Button/Button.style.ts

```diff
@@ -72,67 +75,112 @@ const colorsFromProps = ({ variant }: ButtonStyleProps) => {
   return styles
 }
 
-const sizeFromProps = ({ size = 'regular', full, hasText }: ButtonStyleProps) => {
-  let padding, fontSize
+const sizeFromProps = ({ size = 'medium', full, hasText }: ButtonStyleProps) => {
+  let padding, fontSize, buttonSizeWithoutText
   switch (size) {
-    case 'smaller': {
-      padding = '10px'
+    case 'small': {
+      padding = 'var(--vertical-padding-small) var(--horizontal-padding-small)'
       fontSize = typography.sizes.button.small
+      buttonSizeWithoutText = '32px'
       break
     }
-    case 'small': {
-      padding = hasText ? `12px 14px` : '12px'
+    case 'medium': {
+      padding = hasText ? `var(--vertical-padding-medium) var(--horizontal-padding-medium)` : '0'
       fontSize = typography.sizes.button.medium
+      buttonSizeWithoutText = '40px'
       break
     }
-    case 'regular':
+    case 'large':
     default: {
-      padding = hasText ? `14px 20px` : '14px'
+      padding = hasText ? `var(--vertical-padding-large) var(--horizontal-padding-large)` : '0'
       fontSize = typography.sizes.button.large
+      buttonSizeWithoutText = '48px'
       break
     }
   }
   return css`
-    width: ${full ? '100%' : ''};
+    width: ${!hasText ? buttonSizeWithoutText : full ? '100%' : ''};
```

Comment:
Probably worth adding `min-width` for when we specify the width explicitly, so the buttons don't get squeezed